anilist.py

```diff
@@ -172,54 +172,78 @@ def match_to_plex(anilist_series: List[AnilistSeries], plex_series_watched: List
         if len(plex_seasons) > 1:
-            custom_mapping_season_count = 0
+            anilist_matches = []
             for index, plex_season in enumerate(plex_seasons):
+
                 season_mappings = retrieve_season_mappings(
                     plex_title, plex_season.season_number
                 )
-                matched_id = 0
                 if season_mappings:
                     matched_id = season_mappings[0].anime_id
-                    if custom_mapping_seasons_anilist_id in (0, matched_id):
-                        custom_mapping_season_count += 1
-                        custom_mapping_seasons_anilist_id = matched_id
-
-                        # Check if season should be added together
-                        if (index != 0 and plex_season.first_episode > plex_seasons[index - 1].last_episode):
-                            plex_watched_episode_count_custom_mapping = plex_season.watched_episodes
-                        else:
-                            plex_watched_episode_count_custom_mapping += plex_season.watched_episodes
-
-            # If we had custom mappings for multiple seasons with the same ID use
-            # cumulative episode count and skip per season processing
-            if custom_mapping_season_count > 1:
-                logger.warning(
-                    "[ANILIST] Found same custom mapping id for multiple seasons "
-                    "so not using per season processing but updating as one | "
-                    f"title: {plex_title} | anilist id: {custom_mapping_seasons_anilist_id} | "
-                    f"total watched episodes: {plex_watched_episode_count_custom_mapping}"
+
+                    custom_mapped_seasons.append(plex_season.season_number)
+                    match = next(
+                        (item for item in anilist_matches if item['anilist_id'] == matched_id),
+                        None,
+                    )
+
+                    if not match:
+                        anilist_matches.append({
+                            "anilist_id": matched_id,
+                            "watched_episodes": plex_season.watched_episodes,
+                            "total_episodes": plex_season.last_episode,
+                            "mapped_seasons": [plex_season.season_number],
+                            "duplicate_mappings": 0,
+                        })
+                        continue
+                    # For multiple seasons with the same anilist id check how they should be added together
+                    elif (index != 0 and plex_season.first_episode > plex_seasons[index - 1].last_episode):
+                        match["watched_episodes"] = plex_season.watched_episodes
+                    else:
+                        match["watched_episodes"] += plex_season.watched_episodes
+
+                        logger.warning(f"[MAPPING] Duplicate anilist-id found for custom mapping | "
```

Comment:
I think we can remove this warning, otherwise it creates quite some spam for some series like One Piece.